cmd/backup/script.go

```diff
@@ -477,28 +392,28 @@ func (s *script) encryptArchive() error {
 
 	gpgFile := fmt.Sprintf("%s.gpg", s.file)
 	s.registerHook(hookLevelPlumbing, func(error) error {
-		if err := remove(gpgFile); err != nil {
+		if err := u.Remove(gpgFile); err != nil {
 			return fmt.Errorf("encryptBackup: error removing gpg file: %w", err)
 		}
 		s.logger.Infof("Removed GPG file `%s`.", gpgFile)
 		return nil
 	})
 
 	outFile, err := os.Create(gpgFile)
-	defer outFile.Close()
 	if err != nil {
 		return fmt.Errorf("encryptBackup: error opening out file: %w", err)
 	}
+	defer outFile.Close()
```

Comment:
What's the reason for moving these defer statements?

Reply:
Because of an IDE warning: `should check returned error before deferring outFile.Close() (SA5001) go-staticcheck`

Reply:
Ah ok. It's notoriously hard to know in which cases you need to do it before or after checking the error (sometimes you'll leak even in case of an error), so it's good the static checker figures this out for us.

Reply:
> It's notoriously hard to know in which cases you need to do it before or after checking the error

FWIW, Staticcheck isn't doing any "figuring out" here. It is generally agreed upon that one shouldn't return a value that needs to be closed together with a non-nil error. Any package doing so can probably be considered buggy (or at least unidiomatic). It's worth noting that in the 6 years this check has existed, nobody has complained (to me) about a false positive. On the flip side, a lot of code returns values that _mustn't_ be closed when the error is non-nil.